woocommerce/woocommerce-ios
# [Subscriptions] Display subscriptions in order details

Closes: #8950
⚠️ Depends on #9535


## Description

This PR adds the UI to display a new Subscriptions section in order details, when an order is associated with one or more subscriptions.

### Changes

* Adds a new `OrderSubscriptionTableViewCell` to show the Subscription details. (I considered adding this as a new SwiftUI view in a `HostingTableViewCell`, but that is designed for the table cell to be configured within the view controller. Because of how the order details view is configured, it was more straightforward to use UIKit for the new cell.)
* Adds a `UILabel+SubscriptionStatus` extension to handle the styles for the subscription status label in the subscription cell.
* Updates `OrderDetailsDataSource` to configure the subscription section and cells, when the order has associated subscriptions.

## Testing instructions


### Prerequisites

1. Install and activate the [Subscriptions extension](https://woocommerce.com/products/woocommerce-subscriptions/) on your store.
2. Create at least one subscription product.
3. Create an order for that subscription product, which creates a new subscription.

### To test

1. Add a breakpoint to `OrderDetailsViewModel` on L611.
2. Build and run the app.
3. Go to the Orders tab.
4. Open the subscription order and confirm the Subscription section appears with the expected subscription details.

You can also open a non-subscription order and confirm the subscription section does not appear.

## Screenshots

https://user-images.githubusercontent.com/8658164/233993238-7b4fe750-8452-480c-b6d1-b92fca41c499.mp4





Subscription row|Large font size|Landscape
-|-|-
![Simulator Screenshot - iPhone 14 Pro - 2023-04-24 at 13 15 28](https://user-images.githubusercontent.com/8658164/233993351-ee320757-a1b2-4acf-b23d-aa4cfa543f0c.png)|![Simulator Screenshot - iPhone 14 Pro - 2023-04-24 at 11 41 16](https://user-images.githubusercontent.com/8658164/233993618-796e9b19-06f2-4847-a2d0-139ebc539cd7.png)|![Simulator Screenshot - iPhone 14 Pro - 2023-04-24 at 11 40 43](https://user-images.githubusercontent.com/8658164/233993634-9e804685-4dd3-45c9-83ef-3ad4e0a823dc.png)


---
